### PR TITLE
Fix event category and tag creation summary in CSV imports.

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/105888

This PR updates the `File_Importer` class to keep track of, and add to the activity log, event category and tag creation during CSV imports.
The functionality is restored both in WP-CLI context and in UI context.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $tt_id is never used | [File_Importer.php:290](https://github.com/moderntribe/the-events-calendar/blob/bcc3f15c584256b7d7d7dbfcaa3042f658e7e521/src/Tribe/Importer/File_Importer.php#L290)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Importer/File_Importer.php [hunk 1]
```diff
@@ -378,4 +403,34 @@ protected function get_featured_image( $event_id, array $record ) {
...
+		return $rewind ? $iterator : new NoRewindIterator( $iterator );
+	}
+
+	/**
+	 * Hooks on the term creation to watch for any newly created terms.
+	 *
+	 * @since TBD
+	 */
+	public function watch_term_creation() {
+		add_action( 'created_term', array( $this, 'on_created_term' ), 10, 3 );
```
Comment (sc0ttkclark) [inline src/Tribe/Importer/File_Importer.php]: Should we check if `has_action` before adding it (potentially again)?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $tt_id is never used | [File_Importer.php:399](https://github.com/moderntribe/the-events-calendar/blob/861abd13473e0d189fd3e553541e27c3904854a6/src/Tribe/Importer/File_Importer.php#L399)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Importer/File_Importer_Events.php [hunk 2]
```diff
@@ -90,6 +90,8 @@ protected function match_existing_post( array $record ) {
 	protected function update_post( $post_id, array $record ) {
 		$update_authority_setting = tribe( 'events-aggregator.settings' )->default_update_authority( 'csv' );
 
+		$this->watch_term_creation();
```
Comment (sc0ttkclark) [inline src/Tribe/Importer/File_Importer_Events.php]: Should there be an unwatch (remove_action) call after everything is run in this method?

File: src/Tribe/Importer/File_Importer.php (hunk 1 above)
Comment (lucatume) [reply src/Tribe/Importer/File_Importer.php]: yep, good idea, thanks

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $tt_id is never used | [File_Importer.php:399](https://github.com/moderntribe/the-events-calendar/blob/1d20977d8449907e0843c8c8b8618c13eaf7606d/src/Tribe/Importer/File_Importer.php#L399)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Importer/File_Importer_Events.php (hunk 2 above)
Comment (lucatume) [reply src/Tribe/Importer/File_Importer_Events.php]: Good idea, added.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $tt_id is never used | [File_Importer.php:399](https://github.com/moderntribe/the-events-calendar/blob/9c5ebbd19a4bfc3b93e671c7cc95ddacbbd64db7/src/Tribe/Importer/File_Importer.php#L399)

> via jenkins/codesniffer-phpcs

**Comment by Camwyn** [comment]
@lucatume for ref, you can add `unused_` to the beginning of the param and the bot will ignore it,
so `$unused_tt_id`

**Pull request merged**